pkg/translate/service.go

```diff
@@ -25,41 +23,44 @@ type getServiceParams struct {
 	id uuid.UUID
 }
 
-func (g *getServiceRequest) parseParams() (getServiceParams, error) {
-	if g == nil {
-		return getServiceParams{}, errors.New("request is nil")
-	}
-
-	id, err := uuid.Parse(g.Id)
+func parseGetServiceRequestParams(req *translatev1.GetServiceRequest) (*getServiceParams, error) {
+	id, err := uuidFromProto(req.GetId())
 	if err != nil {
-		return getServiceParams{}, fmt.Errorf("parse uuid: %w", err)
+		return nil, fmt.Errorf("parse id: %w", err)
 	}
 
-	return getServiceParams{id: id}, nil
+	return &getServiceParams{id: id}, nil
 }
 
-func (g *getServiceParams) validate() error {
-	// validate if uuid is in DB
+func validateGetServiceRequestParams(params *getServiceParams) error {
```

Comment:
We can leave it as it was previously - `func (g *getServiceParams) validate() error`.

The benefit of `parseGetServiceRequestParams` is it removes the need for the declaration of `params` in a consumer function, and keeps the consumer function tidier.